sdk/src/main/java/com/mapbox/maps/renderer/FpsManager.kt

```diff
@@ -0,0 +1,208 @@
+package com.mapbox.maps.renderer
+
+import android.os.Handler
+import android.os.Looper
+import androidx.annotation.AnyThread
+import androidx.annotation.WorkerThread
+import androidx.core.os.postDelayed
+import com.mapbox.maps.logE
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import kotlin.math.pow
+
+internal class FpsManager {
+  private var userDefinedRefreshRate = USER_DEFINED_REFRESH_RATE_NOT_SET
+  private var userDefinedRatio = USER_DEFINED_RATIO_NOT_SET
+
+  private var screenRefreshRate = SCREEN_METRICS_NOT_DEFINED
+  private var screenRefreshPeriodNs = -1L
+  private var previousRefreshNs = -1L
+  private var previousDrawnFrameIndex = 0
+  private var frameRenderTimeAccumulatedNs = 0L
+
+  private var preRenderTimeNs = -1L
+  private var choreographerTicks = 0
+  private var choreographerSkips = 0
+
+  @Volatile
+  private var fpsChangedListener: OnFpsChangedListener? = null
+
+  private var handler: Handler? = null
+
+  @WorkerThread
+  fun prepareHandler() {
+    Looper.myLooper()?.let { looper ->
+      handler = Handler(looper)
+    } ?: logW(
+      TAG,
+      "Looper could not be found for Mapbox render thread! " +
+        "Some FPS reports may be inaccurate."
+    )
+  }
+
+  @WorkerThread
+  fun setScreenRefreshRate(screenRefreshRate: Int) {
```

Comment:
with `Int.MAX_VALUE` map breaks

Reply:
good catch, will change boundaries and add code check for range (also will support Java properly then)